manifests/prepare.pp

```diff
@@ -87,12 +87,20 @@
   # the osfamily of the client being configured.
 
   case $::osfamily {
-    'redhat', 'debian', 'windows', 'solaris', 'aix', 'suse', 'darwin': {
+    'redhat', 'debian', 'windows', 'aix', 'suse', 'darwin': {
       class { $_osfamily_class:
         package_file_name => $package_file_name,
       }
       contain $_osfamily_class
     }
+    'solaris': {
+      if $::aio_agent_version != $package_version {
```

Comment:
We could, it was mainly to avoid rerunning the exec statements which are not idempotent with solaris so I figured I'd limit the impact of the change to just solaris.

We were specifically having issues with this exec statement (https://github.com/demophoon/puppetlabs-puppet_agent/blob/0eb56e366c84ebc9d0a720e8a1227cf9f259d7ca/manifests/osfamily/solaris.pp#L22-L28) always trying to untar the agent tarball downloaded from the master. And in the case that the solaris agent is already AIO then we never download a tarball and the exec command fails to run since there isn't anything to untar.